core/services/chainlink/application.go

```diff
@@ -387,7 +387,7 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 		peerWrapper = ocrcommon.NewSingletonPeerWrapper(keyStore, cfg.P2P(), cfg.OCR(), cfg.Database(), db, globalLogger)
 		srvcs = append(srvcs, peerWrapper)
 	} else {
-		globalLogger.Debug("P2P stack disabled")
+		return nil, fmt.Errorf("P2P stack required for OCR or OCR2")
```

Comment:
This works, but I think we can provide improved UX if we check from the TOML config validation, since that is used by the `validate` subcommand too, which will flag problems sooner.